parsl/executors/high_throughput/interchange.py

```diff
@@ -0,0 +1,310 @@
+             Worker url on which workers will attempt to connect back
+
+        heartbeat_period : int
+             Heartbeat period expected from workers
+
+        logging_level : int
+             Logging level as defined in the logging module. Default : logging.INFO (20)
+
+        """
+        start_file_logger("interchange.logs", level=logging_level)
+        logger.debug("****************************************")
+        logger.debug("Starting Interchange process v0.2")
+        logger.debug("****************************************")
+
+        self.client_address = client_address
+        self.interchange_address = interchange_address
+        self.identity = uuid.uuid4()
+
+        logger.info("Attempting connection to client at:{} port:{},{}".format(
+            client_address, client_ports[0], client_ports[1]))
+        self.context = zmq.Context()
+        self.task_incoming = self.context.socket(zmq.DEALER)
+        self.task_incoming.RCVTIMEO = 100  # in milliseconds
+        self.task_incoming.connect("tcp://{}:{}".format(client_address, client_ports[0]))
+        self.results_outgoing = self.context.socket(zmq.DEALER)
+        self.results_outgoing.connect("tcp://{}:{}".format(client_address, client_ports[1]))
+        logger.debug("Connected to client")
+
+        self.worker_ports = worker_ports
+        self.worker_port_range = worker_port_range
+
+        self.task_outgoing = self.context.socket(zmq.ROUTER)
+        self.results_incoming = self.context.socket(zmq.ROUTER)
+
+        if self.worker_ports:
+            self.worker_task_port = self.worker_ports[0]
+            self.worker_result_port = self.worker_ports[1]
+
+            self.task_outgoing.bind("tcp://*:{}".format(self.worker_task_port))
+            self.results_incoming.bind("tcp://*:{}".format(self.worker_result_port))
+
+        else:
+            self.worker_task_port = self.task_outgoing.bind_to_random_port('tcp://*',
+                                                                           min_port=worker_port_range[0],
+                                                                           max_port=worker_port_range[1], max_tries=100)
+            self.worker_result_port = self.results_incoming.bind_to_random_port('tcp://*',
+                                                                                min_port=worker_port_range[0],
+                                                                                max_port=worker_port_range[1], max_tries=100)
+
+        logger.info("Bound to ports:{},{} for incoming worker connections".format(
+            self.worker_task_port, self.worker_result_port))
+
+        self._task_queue = []
+        self._ready_worker_queue = {}
+        self.max_task_queue_size = 10 ^ 5
+
+        self.heartbeat_thresh = heartbeat_period * 2
+
+    def get_tasks(self, count):
+        """ Get's a batch of tasks from the task queue
+
+        Parameters
+        ----------
+             count: int
```

Comment:
Move everything up an indent level (`count` should be same level as `Parameters`)